lib/Runtime/Library/TypedArray.cpp

```diff
@@ -678,32 +714,39 @@ namespace Js
 
         if (index < GetLength())
         {
-            return true;
+            return Property_Found;
         }
-        return false;
+
+        return Property_NotFound_NoProto;
     }
 
-    BOOL TypedArrayBase::GetItem(Var originalInstance, uint32 index, Var* value, ScriptContext* requestContext)
+    PropertyQueryFlags TypedArrayBase::GetItemQuery(Var originalInstance, uint32 index, Var* value, ScriptContext* requestContext)
     {
         *value = DirectGetItem(index);
-        return true;
+        return index < GetLength() ? Property_Found : Property_NotFound_NoProto;
     }
 
-    BOOL TypedArrayBase::GetItemReference(Var originalInstance, uint32 index, Var* value, ScriptContext* requestContext)
+    PropertyQueryFlags TypedArrayBase::GetItemReferenceQuery(Var originalInstance, uint32 index, Var* value, ScriptContext* requestContext)
```

Comment:
var a = // some typed array
a[3];   // GetItemQuery
a[3]();   //  GetItemReferenceQuery


---
In reply to: [106301540](https://github.com/Microsoft/ChakraCore/pull/2640#discussion_r106301540) [](ancestors = 106301540)